src/GraphQL.Tests/Serialization/GraphQLRequestTests.cs

```diff
@@ -0,0 +1,246 @@
+            actual.Query.ShouldBe("hello");
+            Verify(actual.Variables);
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_GraphQLRequest_IsCaseSensitive(IGraphQLTextSerializer serializer)
+        {
+            var test = $"{{\"Query\":\"hello\",\"Variables\":{ExampleJson}}}";
+            Should.Throw<Exception>(() => serializer.Deserialize<GraphQLRequest>(test));
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_GraphQLRequest_List(IGraphQLTextSerializer serializer)
+            => Reads_GraphQLRequest_Test<List<GraphQLRequest>>(serializer);
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_GraphQLRequest_Enumerable(IGraphQLTextSerializer serializer)
+            => Reads_GraphQLRequest_Test<IEnumerable<GraphQLRequest>>(serializer);
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_GraphQLRequest_Array(IGraphQLTextSerializer serializer)
+            => Reads_GraphQLRequest_Test<GraphQLRequest[]>(serializer);
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_GraphQLRequest_IList(IGraphQLTextSerializer serializer)
+            => Reads_GraphQLRequest_Test<IList<GraphQLRequest>>(serializer);
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_GraphQLRequest_IReadOnlyList(IGraphQLTextSerializer serializer)
+            => Reads_GraphQLRequest_Test<IReadOnlyList<GraphQLRequest>>(serializer);
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_GraphQLRequest_ICollection(IGraphQLTextSerializer serializer)
+            => Reads_GraphQLRequest_Test<ICollection<GraphQLRequest>>(serializer);
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_GraphQLRequest_IReadOnlyCollection(IGraphQLTextSerializer serializer)
+            => Reads_GraphQLRequest_Test<IReadOnlyCollection<GraphQLRequest>>(serializer);
+
+        private void Reads_GraphQLRequest_Test<T>(IGraphQLTextSerializer serializer)
+            where T : IEnumerable<GraphQLRequest>
+        {
+            var test = $"[{{\"query\":\"hello\",\"variables\":{ExampleJson}}}, {{\"query\":\"dummy\"}}]";
+            var actual = serializer.Deserialize<T>(test);
+            actual.Count().ShouldBe(2);
+            var request = actual.First();
+            request.Query.ShouldBe("hello");
+            Verify(request.Variables);
+            var request2 = actual.Last();
+            request2.Query.ShouldBe("dummy");
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_GraphQLRequest_List_Multiple_Items(IGraphQLTextSerializer serializer)
```

Comment:
`private void Reads_GraphQLRequest_Test<T>(IGraphQLTextSerializer serializer)` covers this test, doesn't it?

Reply:
Yeah, I had tests written in two different locations and rather than delete some and figure out what was a duplicate and what wasn't, I just made sure they all worked properly and put them all in the same file.